### PR TITLE
V1

### PR DESCRIPTION
**Pull request by orthagonal**

### CODE REVIEW

File: main.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+const async = require('async');
+const path = require('path');
+const fs = require('fs');
+
+const execute = (fileName, argv, callback) => {
+  // establish AWS credentials:
+  const aws = awsAuth(argv);
+  // make sure working directory is present:
+  if (!fs.existsSync) {
+    fs.mkdirSync('build');
```
Comment (jgallen23) [inline main.js]: what is this for?

File: lib/commonAWS.js [hunk 2]
```diff
@@ -0,0 +1,62 @@
+'use strict';
```
Comment (jgallen23) [inline lib/commonAWS.js:1]: think we could pull this out into a separate module?

File: cli.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+const argv = require('yargs')
+.options({
+  profile: {
+    describe: 'name of aws profile to use',
+    default: false
+  },
+  bucket: {
+    describe: 's3 bucket to upload the image to',
+    default: 's3-bucket'
```
Comment (jgallen23) [inline cli.js]: no default.  should be required to set

File: lib/image.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  const handleResult = result => {
+    if (result.length === 0) {
+      return callback('Unable to compress image');
+    }
+    callback(null, result[0].path);
+  };
+  const options = {
+    plugins: [imageminMozjpeg({ quality: level }), imageminPngquant({ quality: level })]
+  };
+  imagemin([file], 'build/', options).then(handleResult);
```
Comment (jgallen23) [inline lib/image.js]: what about building to a temp directory?

**Review by jgallen23 — CHANGES_REQUESTED** [review]
overall, looks good

File: lib/commonAWS.js (hunk 2 above)
Comment (orthagonal) [reply lib/commonAWS.js:1]: Definitely, we are just going to keep reusing this code for our AWS stuff.  Make a repo for it and add a readme with any specs you want and I'll switch over to it.

File: lib/image.js (hunk 4 above)
Comment (orthagonal) [reply lib/image.js]: That's what the 'mkdir' is doing there, making sure the 'build' directory is present where the files are going to be stored.

File: main.js (hunk 1 above)
Comment (orthagonal) [reply main.js]: Imagemin wants to read/write things from file, so I had it make a temp working directory.

File: lib/image.js (hunk 4 above)
Comment (jgallen23) [inline lib/image.js]: that's not really a temp directory. look at os.tmpdir

**Review by jgallen23 — CHANGES_REQUESTED** [review]

File: cli.js (hunk 3 above)
Comment (jgallen23) [reply cli.js]: ping

Comment (orthagonal) [reply cli.js]: Committed the code for this yesterday, but was waiting on other feedback before pushing the commit (which causes Github to hide our comments pertaining to the old commit)

File: lib/image.js (hunk 4 above)
Comment (orthagonal) [reply lib/image.js]: Oh I didn't know about that one, that's very nice.

**Comment by jgallen23** [comment]
Error: Cannot find module 'imagemin-mozjpeg' (I manually installed, but needs to be added to package.json)

**Comment by jgallen23** [comment]
```
20:51 $ node ~/libraries/s3put/cli.js --profile ft --bucket project-hub hero.jpg                                                                        
Unable to locate profile ft in file /home/ubuntu/.aws/config.
Falling back to us-east-1
{ ETag: '"d41d8cd98f00b204e9800998ecf8427e"',
  Location: 'https://project-hub.s3.amazonaws.com/hero.jpg',
  key: 'hero.jpg',
  Key: 'hero.jpg',
  Bucket: 'project-hub' }
```

uploaded a file with 0 bytes

**Pull request merged**